packages/ui/components/ui/tx/view/value.tsx

```diff
@@ -1,21 +1,90 @@
-import { ValueView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { Metadata, ValueView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
 import {
   fromBaseUnitAmount,
   getDisplayDenomExponent,
   getDisplayDenomFromView,
 } from '@penumbra-zone/types';
 import { CopyToClipboard } from '../../copy-to-clipboard';
+//import { AssetIcon } from '../../webapp/shared/asset-icon.tsx';
 import { CopyIcon } from '@radix-ui/react-icons';
 import { Amount } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/num/v1/num_pb';
+import { Identicon } from '../../identicon';
 
 interface ValueViewProps {
   view: ValueView | undefined;
   showDenom?: boolean;
+  showValue?: boolean;
+  showIcon?: boolean;
+  showEquivalent?: boolean;
 }
 
-export const ValueViewComponent = ({ view, showDenom = true }: ValueViewProps) => {
+export const AssetIcon = ({ metadata }: { metadata?: Metadata }) => {
+  const icon = metadata?.images[0]?.png || metadata?.images[0]?.svg;
+  return (
+    <>
+      {icon ? (
+        <img className='size-6 rounded-full' src={icon} alt='Asset icon' />
+      ) : (
+        <Identicon
+          uniqueIdentifier={metadata?.symbol ?? '?'}
+          size={24}
+          className='rounded-full'
+          type='solid'
+        />
+      )}
+    </>
+  );
+};
```

Comment:
I didn't know how to deduplicate this with the other `AssetIcon` or move it between packages. Anyways, the existing one is wrongly defined now that we have images in the metadata, there's no need for a memoized search.